virtualmachine/gcp/vm.go

```diff
@@ -56,17 +65,127 @@ type VM struct {
 
 	AccountFile  string
 	account      accountFile
-	SSHCreds     ssh.Credentials // privateKey is required for GCE
+	SSHCreds     ssh.Credentials
 	SSHPublicKey string
+
+	Firewall  string // required when modifying firewall rules
+	Endpoints []Endpoint
+}
+
+// Endpoint represents the protocol and ports configured in a firewall
+type Endpoint struct {
+	// Protocol can be one of the following well known protocol strings
+	// (tcp, udp, icmp, esp, ah, sctp) which are supported by GCP
+	Protocol string
+	Ports    []string
+}
+
+// Image represents a GCE image
+type Image struct {
+	Id                uint64 `json:"id,omitempty,string"`
+	CreationTimestamp string `json:"creation_timestamp,omitempty"`
+	// DeprecationStatus indicates whether the image is depcrecated. If the
+	// image is not deprecated it will have an empty string.
+	DeprecationStatus string `json:"deprecation_status,omitempty"`
+	// DeprecationReplacement provides the url of the image which is a
+	// replacement of this deprecated image. If this image is not deprecated
+	// it's value will be an empty string.
+	DeprecationReplacement string `json:"deprecation_replacement,omitempty"`
+	Description            string `json:"description,omitempty"`
+	DiskSizeGb             int64  `json:"disk_size_gb,omitempty,string"`
```

Comment:
should it be type float? or in MBs

Reply:
GCP SDK defines it in GB with int64. For the compatibility, it is kept as int64.